src/main/java/edu/ucsb/cs156/courses/controllers/CoursesController.java

```diff
@@ -0,0 +1,177 @@
+
+        return courses;
+    }
+
+
+    @ApiOperation(value = "Create a new course")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+    public Courses postCourses(
+            @ApiParam("enrollCd") @RequestParam String enrollCd,
+            @ApiParam("psId") @RequestParam Long psId) {
+        CurrentUser currentUser = getCurrentUser();
+        log.info("currentUser={}", currentUser);
+
+        Courses courses = new Courses();
+        courses.setUser(currentUser.getUser());
+        courses.setEnrollCd(enrollCd);
+        courses.setPsId(psId);
+        Courses savedCourses = coursesRepository.save(courses);
+        return savedCourses;
+    }
+
+    @ApiOperation(value = "Delete a course (admin)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("/admin")
+    public Object deleteCourses_Admin(
+            @ApiParam("id") @RequestParam Long id) {
+              Courses courses = coursesRepository.findById(id)
+          .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+
+          coursesRepository.delete(courses);
+
+        return genericMessage("Courses with id %s deleted".formatted(id));
+    }
+
+    @ApiOperation(value = "Delete a course (user)")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @DeleteMapping("/user")
+    public Object deleteCourses(
+            @ApiParam("id") @RequestParam Long id) {
+        User currentUser = getCurrentUser().getUser();
+        Courses courses = coursesRepository.findByIdAndUser(id, currentUser)
+          .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+        coursesRepository.delete(courses);
+        return genericMessage("Courses with id %s deleted".formatted(id));
+    }
+
+    @ApiOperation(value = "Update a single Course (admin)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public Courses putCourseById_admin(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Courses incomingCourses) {
+              Courses courses = coursesRepository.findById(id)
+          .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+
+          courses.setEnrollCd(incomingCourses.getEnrollCd());
+          courses.setPsId(incomingCourses.getPsId());
+
+        coursesRepository.save(courses);
+
+        return courses;
+    }
+
+    @ApiOperation(value = "Update a single course (user)")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("/user")
+    public Courses putCoursesById(
+            @ApiParam("id") @RequestParam Long id,
+            @RequestBody @Valid Courses incomingCourses) {
+        User currentUser = getCurrentUser().getUser();
+        Courses courses = coursesRepository.findByIdAndUser(id, currentUser)
+          .orElseThrow(() -> new EntityNotFoundException(Courses.class, id));
+
+        courses.setEnrollCd(incomingCourses.getEnrollCd());
+        courses.setPsId(incomingCourses.getPsId());
+
```

Comment:
Just like in create, you need to verify that the psId matches the psId of an existing personal scheudle, that it is for the correct user, and that the enrollCd exists for that quarter.  Otherwise you need to throw an error.